### PR TITLE
Add AWS IoT websocket URL presigner

### PR DESCRIPTION
**Pull request by at-wat**

### CODE REVIEW

**Comment by at-wat** [comment]
@t-asaka @jojo43 please take a look.
I confirmed that the wss URL generated from this signer could establish a connection to AWS IoT.

File: presigner/presign.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	_, err = signer.Presign(
+		req, body,
+		a.clientConfig.SigningName, a.clientConfig.SigningRegion,
+		expire, from,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return req.URL.String() + "&X-Amz-Security-Token=" + url.QueryEscape(sessionToken), nil
```
Comment (t-asaka) [inline presigner/presign.go]: I think that it is better to add it only when sessionToken exists.

File: presigner/presign_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func ExamplePresigner_PresignWss() {
```
Comment (at-wat) [inline presigner/presign_test.go]: it seems better to be normal test instead of example test

File: presigner/presign.go (hunk 1 above)
Comment (at-wat) [reply presigner/presign.go]: fixed and added a test for it

**Pull request merged**